lite2/client.go

```diff
@@ -757,6 +782,65 @@ func (c *Client) fetchHeaderAndValsAtHeight(height int64) (*types.SignedHeader,
 	return h, vals, nil
 }
 
+// fetchMissingTrustedHeader finds the closest height after the
+// requested height and does backwards verification.
+func (c *Client) fetchMissingTrustedHeader(height int64, now time.Time) (*types.SignedHeader, error) {
+	c.logger.Info("Fetching missing header", "height", height)
+
+	trustedHeader, err := c.trustedStore.SignedHeaderAfter(height)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't get signed header after %d", height)
+	}
+
+	var untrustedHeader *types.SignedHeader
+	for i := trustedHeader.Height - 1; i >= height; i-- {
```

Comment:
Would you want the actual backwards verification process to be encapsulated as a separate function. (perhaps in the `verifier.go` file)